documents.go

```diff
@@ -47,3 +60,68 @@ func (cl *DocumentsClientGetAll) Parent(id DocumentID) *DocumentsClientGetAll {
 func (cl *DocumentsClientGetAll) Do(ctx context.Context, fn func(*Document, error) bool) error {
 	return nil
 }
+
+// Create returns a client for creating a single document in the specified collection.
+// collectionID: The ID of the collection to which the document will be added.
+// title: The title of the document (optional, set to empty string to omit).
+// text: The content text of the document.
+// publish: A boolean indicating whether the document should be published.
+func (cl *DocumentsClient) Create(collectionID CollectionID, title string, text string, publish bool, parentDocumentId ParentDocumentID, templateId TemplateID, template bool) *DocumentsClientCreate {
```

Comment:
Do we already want to go with such a complicated strucutre?
We need it earlier or later anyways, but so far we just went with the basics `create collection(name)`, that's it.
Just wanted to questions this, no changes requested.

Also, is it common to have so many parameters in go (in a single line)? 🤔 
If is so, I like the Kotlin code style to have each param in its own line:
```go
func (cl *DocumentsClient) Create(
    collectionID CollectionID, 
    title string,
    text string,
    ....,
```
But I have no strong opinion on that and don't know whats "the way" to go.
@rsjethani ? 🙂

Reply:
Do we already want to go with such a complicated strucutre?
- We (Sre) have a ticket, which needs us to be able to create a document with these features (only template + templateID where added as an extra, since it would be half implemented anyways)

Multiline parameters:
No Idea, I'd also be in favor of the kotlin way :)

Reply:
@zadjadr 
Multi-line parameters are usually not preferred, they are usually a code smell hinting that code restructuring can reduce number of parameters required. A very straight forward approach is to only ask for mandatory parameters. In this case you should take only `title` and `collection id` as compulsory values. For optional values you can have `option methods` which configure optional behavior. BTW you cannot have create method on DocumentsClient, you need a new type `DocumentsClientCreate` to maintain consistency across api usage. Following is how the API call can look like eventually:
```
client.Documents().Create(title, collection id).Text("doc text").Do(...)
```
Observe: `Text` is an example of a option method which allows configuring optional behavior. If we do not want to set text for the document then that call can be omitted.